src/plugins/identity/README.md

```diff
@@ -10,10 +10,132 @@ GitHub and Discourse (potentially with a different username on each service).
 We would like to combine these two identities together, so that we can
 represent that user's combined cred properly. The Identity plugin enables this.
 
-Specifically, the instance maintainer can provide a (locally unique) username
-for the user, along with a list of aliases the user is known by, e.g.
-`github/username` and `discourse/other_username`. The aliases are simple string
-representations, that are intended to be easy to maintain by hand in a
-configuration file. Then, the identity plugin will provide a list of
-`NodeContraction`s that can be used by `Graph.contractNodes` to combine the
-user identities as described.
+## Status and Caveats
+
+The Identity plugin currently only applies to new (Weighted)Graphs and Cred
+scores being generated. When including historical Graphs/Cred scores, for
+example to distribute Grain, you may need to contract the identities again.
+This can be tricky to do correctly, as they may have already received Grain
+on their GitHub/Discourse addresses before contraction.
+
+Code-reviewed support for Grain built into SourceCred should handle this in
+the future. But for now, requires the instance maintainer to handle Grain
+account contraction.
+
+## Configuration
+
+Identities can be configured using the `project.json` file.
+By adding entries to the the `identities` array.
+
+As an example of what this would look like:
+
+```js
+[
+  {
+    "type": "sourcecred/project",
+    "version": "0.4.0",
+  },
+  {
+    // The identities we should collapse.
+    "identities": [
+      {
+        "username": "User-A",
+        "aliases": ["github/user-a", "discourse/user-a"],
+      },
+      {
+        "username": "User-B",
+        "aliases": ["github/user-b", "discourse/user-b"],
+      },
+    ],
+
+    "id": "@example-project",
+    "discourseServer": {
+      // ...
+    },
+    "repoIds": [
+      // ...
+    ],
+  },
+]
+```
+
+### Username
+
+The `username` field allows you to choose a new username for the collapsed
+identity.
+
+It must be _unique in this project_ and can contain `A-Z`, `a-z`, `0-9`, `-` or
+`_` as valid characters.
+
+For display purposes the username is prefixed with an `@`.
+If a username is prefixed with `@` in the project file, it will be ignored.
+Meaning `"username": "User-A"` and `"username": "@User-A"` are equivalent.
+
+### Aliases
+
+Aliases have a format of:
+`{plugin}/{user ID for this plugin}`
+
+Currently supported plugins are:
+
+| Plugin | Example alias | Notes |
+|---|---|---|
+| GitHub | `github/example-user` | A user account on GitHub's platform. |
+| Discourse | `discourse/example-user` | A user account on the Discourse server configured in this project. |
+
+
+## How a contraction works
+
+Firstly, contractions are done _after_ the plugins have created their repective
```

Comment:
repective-> respective

Reply:
Thanks, forgot to run a spell check. Fixed.